src/Components/Endpoints/src/Rendering/EndpointHtmlRenderer.Streaming.cs

```diff
@@ -162,8 +162,34 @@ private void WriteComponentHtml(int componentId, TextWriter output, bool allowBo
     {
         _visitedComponentIdsInCurrentStreamingBatch?.Add(componentId);
 
-        var renderBoundaryMarkers = allowBoundaryMarkers
-            && ((EndpointComponentState)GetComponentState(componentId)).StreamRendering;
+        var componentState = (EndpointComponentState)GetComponentState(componentId);
+        var renderBoundaryMarkers = allowBoundaryMarkers && componentState.StreamRendering;
+
+        // TODO: It's not clear that we actually want to emit the interactive component markers using this
+        // HTML-comment syntax that we've used historically, plus we likely want some way to coalesce both
+        // marker types into a single thing for auto mode (the code below emits both separately for auto).
+        // It may be better to use a custom element like <blazor-component ...>[prerendered]<blazor-component>
+        // so it's easier for the JS code to react automatically whenever this gets inserted or updated during
+        // streaming SSR or progressively-enhanced navigation.
```

Comment:
With this PR I'm not trying to express a final opinion on the exact format of the interactive component markers that we output. We can determine that when implementing the client-side pieces, based on what's best for the client-side code (e.g., whether it should change from being a comment tag to a custom element).

The goal for this PR is just to emit the output in some sensible format. The specific format used in this PR happens to be exactly what we already output, and hence is compatible with `blazor.server.js` etc.

Reply:
The issue with emitting a custom element is that it will affect the layout, isn't it?

Reply:
Yes that is a risk, and is why we haven't done it before. It is a bit of an edge case though, since `<my-custom-container>some markup</my-custom-container>` will *normally* render exactly the same as `some markup`, if there are no CSS rules associated with `my-custom-container`. The cases where it would make a different would be:

 * if you have very specific CSS rules that try to locate things in `some markup` based on them being immediate children of some other element
 * or if the parent is a special element type like `table` that can only have certain child types
 * There may be other such cases too.

We *could* decide that rendermode boundaries are rare enough that we are willing to tell people not to place them at critical locations (like as immediate children of `<table>`), and not to write CSS rules that are so specific in this case.

It's a tradeoff between compatibility and simplicity of implementation, and warrants some open discussion. Overall I suspect we'll end up saying we want to maximize compatibility even if it makes our implementation harder. But we are free to discuss this and be open to other outcomes.